ldes-server-infra-postgres/postgres-pagination-repository/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/pagination/postgres/batch/PageAssigner.java

```diff
@@ -22,9 +23,9 @@ public PageAssigner(DataSource dataSource) {
 	}
 
 	@Override
-	public void write(Chunk<? extends Page> chunk) throws Exception {
-		for (var page : chunk) {
-			jdbcTemplate.update(SQL, page.getId(), page.getBucketId(), page.getAvailableMemberSpace());
+	public void write(Chunk<? extends List<Page>> chunk) throws Exception {
+		for (var pages : chunk) {
+			pages.forEach(page -> jdbcTemplate.update(SQL, page.getId(), page.getBucketId(), page.getAssignedMemberCount()));
 		}
```

Comment:
Maybe instead of using a nested for(each) loop, jdbcTemplate.batchUpdate could be used in the first for-loop